pkg/query-service/app/server.go

```diff
@@ -257,7 +258,15 @@ func (s *Server) Start() error {
 			zap.S().Error("Could not start HTTP server", zap.Error(err))
 		}
 		s.unavailableChannel <- healthcheck.Unavailable
+	}()
+
+	go func() {
+		zap.S().Info("Starting pprof server", zap.String("addr", constants.DebugHttpPort))
 
+		err = http.ListenAndServe(constants.DebugHttpPort, nil)
```

Comment:
Where are we defining/initializing pprof?

Reply:
> _ "net/http/pprof" // http profiler
We import pprof using the above import statement for the side effect of registering its HTTP handlers.

And when not using `DefaultServeMux`, we would have to register handlers with the mux being used or else we can just pass `nil` instead.

Reference: https://pkg.go.dev/net/http/pprof